test/java/integration/AndroidEnvoyExplicitFlowTest.java

```diff
@@ -323,6 +323,7 @@ public MockResponse dispatch(RecordedRequest recordedRequest) {
                                           .setHttpMethod(RequestMethod.POST)
                                           .setUrl(mockWebServer.url("post/flowers").toString())
                                           .addHeader("content-length", "23")
+                                          .addHeader("socket-tag", "0,0")
```

Comment:
wait is the interface for adding these headers from Java?  I thought that it was something like the cleartext or proxy things where android would inform what a given tag should be?

Reply:
As I understand it, the application specifies the tag/uid and then calls:
https://developer.android.com/reference/android/net/TrafficStats#setThreadStatsUid(int)
https://developer.android.com/reference/android/net/TrafficStats#setThreadStatsTag(int)
So in EM, I would expect the application (at the java layer) to set the headers and then this filter will apply them to the socket. Does that sound right?

Reply:
is every application supposed to do this, or is it an opt-in sort of thing?  If everyone is expected to do it I'm not sure how we'd get tags if using the C++ APIS, and I think we have to call up at some point to get the ID.  If it's not used by everyone, I wonder if we should make it off by default and add a config knob to turn it on for cronvoy?

Reply:
As I understand it, this is to be used when an app is doing networking on behalf of another app, or vice versa. So if an app is not doing that, then it would not need to use this Android API. Cronet exposes an API which makes use of this behavior via: https://chromium.googlesource.com/chromium/src/+/refs/heads/main/components/cronet/android/api/src/org/chromium/net/ExperimentalUrlRequest.java#52

So making this off by default and adding a knob that cronvoy turns on seems fine. That being said, I'm not sure there's much harm in making it always on, but happy to make it off by default. Do you have any pointers to how to make such a knob work?

Reply:
I'd be mildly inclined to not have it on by default, or built in by default.  It's already possible to add custom filters - for example I took the "custom" brotli filter out of the cronet tests when I added explicit configuration support for brotli here:
https://github.com/envoyproxy/envoy-mobile/pull/2342/files#diff-3a624bebe9faf2818e3aaeec574223c875803bdbcb6bc2e36c890ce944a3b337L32

I would say we could skip having an explicit knob, but expose default filter config via insert like this:
https://github.com/envoyproxy/envoy-mobile/pull/2342/files#diff-90bc04f1f5d0a531db783375205a81872a6bb7830871d24a01ea1678e6a8793bR161

and then turn it up that way once we do tests.  WDYT?

Reply:
Done, I think!